src/Polly.Specs/RetrySpecs.cs

```diff
@@ -363,5 +363,39 @@ public void Should_create_new_context_for_each_call_to_policy()
 
             contextValue.Should().Be("new_value");
         }
+
+        [Fact]
+        public void Should_not_call_onretry_when_retry_count_is_zero_without_context()
+        {
+            string onRetrySideEffect = "original_value";
```

Comment:
@nelsonghezzi In these tests, do you think it may be more direct to use `bool onRetryInvoked = false;` ,  and have the onRetry delegate `=> { onRetryInvoked = true; };` etc?  A small change, but that way we could avoid the magic strings "original_value" and "new_value"...

And: many thanks once again for your contribution! (A number of people asked for this feature...)